clerk/sessions_test.go

```diff
@@ -242,6 +243,47 @@ func TestSessionsService_Verify_invalidServer(t *testing.T) {
 	}
 }
 
+func TestSessionsService_CreateTokenFromTemplate_Success(t *testing.T) {
+	token := "token"
+	sessionId := fmt.Sprintf("sess_%s", "2bp7pawkvFUR3m3QPBm6Cwx5ghZ")
+	templateSlug := "someTemplateSlug"
+	sessionToken := "someSessionToken"
+	expectedResponse := fmt.Sprintf(dummySessionTokenJsonTemplate, sessionToken)
+
+	clerkClient, mux, _, teardown := setup(token)
+	defer teardown()
+
+	mux.HandleFunc(fmt.Sprintf("/%s/%s/token/%s", SessionsUrl, sessionId, templateSlug), func(w http.ResponseWriter, req *http.Request) {
+		testHttpMethod(t, req, "POST")
+		testHeader(t, req, "Authorization", fmt.Sprintf("Bearer %s", token))
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write([]byte(expectedResponse))
+	})
+
+	var expectedSessionToken SessionToken
+	err := json.Unmarshal([]byte(expectedResponse), &expectedSessionToken)
+	assert.NoError(t, err)
+
+	received, err := clerkClient.Sessions().CreateTokenFromTemplate(sessionId, templateSlug)
+	equal := reflect.DeepEqual(*received, expectedSessionToken)
+	assert.NoError(t, err)
+	assert.Truef(
+		t,
+		equal,
+		"Response = %v, want %v", *received, expectedSessionToken,
+	)
```

Comment:
I believe we can assert that the response is correctly parsed as well. How's this for an assertion?

```go
got, err := clerkClient.Sessions().CreateTokenFromTemplate(sessionId, templateSlug)
require.NoError(t, err)
require.Equal(t, "token", got.Object)
require.Equal(t, sessionToken, got.JWT)
```

Reply:
You are right! :smile: